homeassistant/components/roon/media_player.py

```diff
@@ -0,0 +1,521 @@
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        return self._media_image_url
+
+    @property
+    def media_position(self):
+        """Return position currently playing."""
+        return self._media_position
+
+    @property
+    def media_duration(self):
+        """Return total runtime length."""
+        return self._media_duration
+
+    @property
+    def volume_level(self):
+        """Return current volume level."""
+        return self._volume_level
+
+    @property
+    def is_volume_muted(self):
+        """Return mute state."""
+        return self._is_volume_muted
+
+    @property
+    def volume_step(self):
+        """.Return volume step size."""
+        return self._volume_step
+
+    @property
+    def supports_standby(self):
+        """Return power state of source controls."""
+        return self._supports_standby
+
+    @property
+    def state(self):
+        """Return current playstate of the device."""
+        return self._state
+
+    @property
+    def is_nowplaying(self):
+        """Return true if an item is currently active."""
+        return self.state == STATE_PLAYING
+
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self._source
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._sources
+
+    @property
+    def shuffle(self):
+        """Boolean if shuffle is enabled."""
+        return self._shuffle
+
+    @property
+    def repeat(self):
+        """Boolean if repeat is enabled."""
+        return self._repeat
+
+    def media_play(self):
+        """Send play command to device."""
+        self._server.roonapi.playback_control(self.output_id, "play")
+
+    def media_pause(self):
+        """Send pause command to device."""
+        self._server.roonapi.playback_control(self.output_id, "pause")
+
+    def media_play_pause(self):
+        """Toggle play command to device."""
+        self._server.roonapi.playback_control(self.output_id, "playpause")
+
+    def media_stop(self):
+        """Send stop command to device."""
+        self._server.roonapi.playback_control(self.output_id, "stop")
+
+    def media_next_track(self):
+        """Send next track command to device."""
+        self._server.roonapi.playback_control(self.output_id, "next")
+
+    def media_previous_track(self):
+        """Send previous track command to device."""
+        self._server.roonapi.playback_control(self.output_id, "previous")
+
+    def media_seek(self, position):
+        """Send seek command to device."""
+        self._server.roonapi.seek(self.output_id, position)
+
+    def set_volume_level(self, volume):
+        """Send new volume_level to device."""
+        volume = int(volume * 100)
+        try:
+            self._server.roonapi.change_volume(self.output_id, volume)
+        except Exception as exc:  # pylint: disable=broad-except
```

Comment:
Can we catch a library specific exception instead?

Reply:
I've caught this in a new version of the library. It doesn't need to cascade up to HA.